ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -106,34 +123,40 @@ sub Configure_Cache {
         $self->warn("Cannot determine current AFS cache size, changing only config file");
     }
 
-    my $afs_cacheinfo_fh = CAF::FileEditor->new( $AFS_CACHEINFO, log => $self ); # TODO use FileReader
-    $afs_cacheinfo_fh->cancel();
-    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+)$; ) {
-        $file_afsmount   = $1;
-        $file_cachemount = $2;
+    my $afs_cacheinfo_fh = CAF::FileReader->new( $AFS_CACHEINFO, log => $self );
+    $self->debug(2, "$AFS_CACHEINFO current contents: >>>$afs_cacheinfo_fh<<<");
+    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+|AUTOMATIC)$; ) {
+        $file_afsmount   = $1 if $file_afsmount eq "";
+        $file_cachemount = $2 if $file_cachemount eq "";
         $file_cache      = $3;
-    }
-    else {
+    } elsif ( "$afs_cacheinfo_fh" ne "" ) {
         $self->error("Cannot parse stored AFS cache mount or size from $AFS_CACHEINFO");
         return 1;
     }
     $afs_cacheinfo_fh->close();
 
-    # sanity check - don't allow cachesize bigger than 95% of a partition size
-    $proc = CAF::Process->new( [ "df", "-k", $file_cachemount ], log => $self, keeps_state => 1 );
-    foreach my $line ( split ( "\n", $proc->output() ) ) {
-        if ($line =~ m{^.*?\s+(\d+)\s+\d+\s+\d+\s+\d+%\s+(.*)}) {
-            my $disk_cachesize = $1;
-            my $mount          = $2;
-            if ( $mount eq $file_cachemount && $new_cache > 0.95 * $disk_cachesize ) {
-                $self->error("Cache size ($disk_cachesize) on $mount cannot exceed 95% of its partition size. Not changing.");
-                return 1;
+    if ( $new_cache ne "AUTOMATIC" ) {
+        # sanity check - don't allow cachesize bigger than 95% of a partition size
+        $proc = CAF::Process->new( [ "df", "-k", $file_cachemount ], log => $self, keeps_state => 1 );
+        foreach my $line ( split ( "\n", $proc->output() ) ) {
+            if ($line =~ m{^.*?\s+(\d+)\s+\d+\s+\d+\s+\d+%\s+(.*)}) {
+                my $disk_cachesize = $1;
+                my $mount          = $2;
+                if ( $mount eq $file_cachemount && $new_cache > 0.95 * $disk_cachesize ) {
+                    $self->error("Cache size ($disk_cachesize) on $mount cannot exceed 95% of its partition size. Not changing.");
+                    return 1;
+                }
             }
         }
     }
 
     # adjust stored cache size (gets overwritten on restart for OpenAFS-1.4)
-    if ( $new_cache != $file_cache ) {
+    # Force string interpolation of cache size as it can be AUTOMATIC
+    if ( $file_afsmount eq "" ) {
+        $self->debug(1, "AFS mount point not defined: using default ($AFS_MOUNTPOINT_DEF)");
+        $file_afsmount = $AFS_MOUNTPOINT_DEF;
+    }
+    if ( "$new_cache" ne "$file_cache" ) {
```

Comment:
with this check, the `AFS_CACHEINIFO` is only updated is `file_cache` is modified, nit `file_afsmount` or `file_cache_mount`
i would always write the file, and with the `close` returns a change in content, do the block below.